tests/lib/thrift/muzic/src/integration.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <zlib.h>
+
+enum { test_max_sample_size = 65536 };
+
+static unsigned char defl_buf[test_max_sample_size];
+static unsigned char infl_buf[test_max_sample_size];
+
+static const char sample[]
+    = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do "
+      " eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, "
+      " quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo conse quat.  "
+      "Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat"
+      " nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui    "
+      "officia deserunt mollit anim id est laborum.";
+
+static z_stream* deflater;
+static z_stream* inflater;
+
+static void setup() {
+  deflater = (z_stream*)malloc(sizeof(z_stream));
```

Comment:
This is a really strange style choice for C (i.e. not C++) code, but I guess we can tolerate it for this specific module.